ecosystem/sep-0010.md

```diff
@@ -149,13 +163,17 @@ To validate the challenge transaction the following steps are performed by the s
   * verify that client signature is correct for the master key of the account address;
 * verify that transaction sequenceNumber is equal to zero;
 * use operations's source account to determine the authenticating client and perform any additional service-specific validations.
+* if the transaction has a memo:
+  * verify the transaction's memo and source account can be used to internally identify the authenticated user;
 
 The verification process confirms that a user holds an account. Depending on your application this may mean complete signing authority, some threshold of control, or being a signer of the account. See [Verification](#verification) for examples.
 
 Upon successful verification, service responds with a session JWT, containing the following claims:
 
 * `iss` (the principal that issued a token, [RFC7519, Section 4.1.1](https://tools.ietf.org/html/rfc7519#section-4.1.1)) — a [Uniform Resource Identifier (URI)] for the issuer (`https://example.com` or `https://example.com/G...`)
-* `sub` (the principal that is the subject of the JWT, [RFC7519, Section 4.1.2](https://tools.ietf.org/html/rfc7519#section-4.1.2)) — the public key of the authenticating Stellar account (`G...`)
+* `sub` (the principal that is the subject of the JWT, [RFC7519, Section 4.1.2](https://tools.ietf.org/html/rfc7519#section-4.1.2)) — the public key of the authenticating Stellar account potentially contcatenated with the `memo` value originally specified by the client using an `-` character
+  * For example, when `account` is pooled, the user can be identified using the 32-byte hash memo: `GAMJ4W6QDEKJWZRMAWGD7ETKPIMKUX4NCZ3JBEYPILZSOWTW7NL6TDQ4-yQ2eD5xHRSWUFJfCZqggCS5peHFkb9SXJ3hBjjZjHnc=`
```

Comment:
I agree we don't know yet if there will be more effort immediately put into rolling out multiplexed accounts in all areas of the SDKs, Horizon, etc, however I think if we are willing to limit pooled account memo support here to int64s then we may as well leverage [SEP-23](https://github.com/stellar/stellar-protocol/blob/master/ecosystem/sep-0023.md) and use the `M` strkey format for a multiplexed address as the format that we encode the account and the memo ID.

If multiplexed addresses get their moment then we'll already be using the appropriate address format here.

In the event multiplexed addresses never take off in other ways using the `M` strkey encoding here won't be any worse than us creating our own format.

By using the `M` strkey format we avoid adding a new format here that is unique to SEP-10.

@JakeUrban @tomerweller @ire-and-curses Thoughts?

Reply:
I'm concerned the presence of `M` address in SEP-10 will imply wider ecosystem support. Won't developers be confused to see a new type of Stellar account without being able to query Horizon for its `/account` JSON? I don't want to conflate the _need_ to encode a memo with the _desire_ to use `M` addresses.

Reply:
I think the fact the M accounts cannot be queried via Horizon endpoints is what allows us to use it in a limited capacity. It will be no different if someone tries to query Horizon with `GAMJ4W6QDEKJWZRMAWGD7ETKPIMKUX4NCZ3JBEYPILZSOWTW7NL6TDQ4-yQ2eD5xHRSWUFJfCZqggCS5peHFkb9SXJ3hBjjZjHnc=`, they'll see the same error.

Reply:
What do we gain from using the M-address format? Memos are the tried and true solution to this problem without M-addresses, and without full M-address support in Horizon/SDK's I don't see a reason to use them.

Reply:
I can also see future issues where the `M`-addresses used here complicate the process for full `M`-address support. For example, how will the `memo` value be encoded into the `G`-address to get an `M`-address? It will likely be a different algorithm than the algorithm used to generate real `M`-addresses.